### PR TITLE
Implement ChromaticSum._shoot

### PR DESCRIPTION
**Pull request by rmjarvis**
In PR #1100, I implemented the applyTo method for chromatic objects, so they can be used as photon_ops.  This is now the canonical way to do photon shooting with chromatic objects, since it can use each photon's specific wavelength to do wavelength-dependent things.

One that I skipped in that PR was ChromaticSum.  I suspect I thought that class would only be used for things like bulge + disk, where each object has a spectral SED.  These can't be used as photon_ops, so I didn't implement that functionality for them.

However, @matroxel wants to use ChromaticSum for the RomanPSF to smoothly interpolate across a sensor using a weighted sum of 4 PSFs, corresponding to the 4 corners of the sensor.  So this PR rectifies my oversight, and implements ChromaticSum._shoot, which is the back end function for all PSF applyTo calls.

Note: when implementing this, I added PhotonArray._copyFrom, which can copy portions of the data from one PhotonArray to another.  It copies all the allocated arrays, using an arbitrary mask in both the original and target PhotonArray.  Similar to `assignAt` but more flexible.  (And `assignAt` now uses `_copyFrom` for its implementation.)

I now use this all the places where we were doing this.  I think this is safer than what we were doing, since some PSFs may need things like the pupil positions or time and when generating the positions/fluxes, they may have generated such things.  So it's a little bit less efficient in most cases (but not much) in the interest of avoiding potentially subtle bugs that may turn up in various edge cases.  (We don't currently have any unit tests that care about such things.)

### CODE REVIEW

File: galsim/photon_array.py [hunk 1]
```diff
@@ -205,7 +205,13 @@ def _fromArrays(
         ret._pupil_u = pupil_u
         ret._pupil_v = pupil_v
         ret._time = time
-        ret._is_corr = is_corr
+        ret._is_corr = False
+        if is_corr:
+            from .deprecated import depr
+            depr('is_corr=True', 2.5, '',
```
Comment (jmeyers314) [inline galsim/photon_array.py:211]: What was the former use case and why is it no longer necessary?

File: tests/test_photon_array.py [hunk 2]
```diff
@@ -243,6 +233,69 @@ def test_photon_array():
...
+    assert pa2.pupil_u[47] != pa1.pupil_u[18]
+    assert pa2.pupil_v[47] != pa1.pupil_v[18]
+    assert pa2.time[47] != pa1.time[18]
+
+    # Error if indices are invalid
+    assert_raises(ValueError, pa2.copyFrom, pa1, slice(50,None), slice(50,None))
+    assert_raises(ValueError, pa2.copyFrom, pa1, 100, 0)
+    assert_raises(ValueError, pa2.copyFrom, pa1, 0, slice(None))
+    assert_raises(ValueError, pa2.copyFrom, pa1)
+
```
Comment (jmeyers314) [inline tests/test_photon_array.py:298]: Should probably add some tests for numpy fancy indexing too.
I.e., `pa1.copyFrom(pa2, pa2.flux>10)` and `pa1.copyFrom(pa2, np.where(pa2.flux>10))`

Comment (jmeyers314) [reply tests/test_photon_array.py:298]: And maybe even discontiguous slices.  `pa1.copyFrom(pa2, slice(0, 100, 2))`

**Review by jmeyers314 — APPROVED** [review]
Looks good to me.  I have one question and one additional testing suggestion.

File: galsim/photon_array.py (hunk 1 above)
Comment (rmjarvis) [reply galsim/photon_array.py:211]: It was for the Sum._shoot implementation.  That had done all the photons in one component first, then all the ones in the other component.  I switched that to do the same algorithm I used for ChromaticSum.  This is potentially important if the PSF is time dependent for instance (e.g. includes a PhasePSF), so the order in the PhotonArray matters.  It seems like there are subtle implications to letting them be correlated like that, and it was very easy to switch it to not ever be correlated in the first place.

File: tests/test_photon_array.py (hunk 2 above)
Comment (rmjarvis) [reply tests/test_photon_array.py:298]: Yep.  Good ideas.  Will do.

**Comment by rmjarvis** [comment]
Question for you.  Do you think we should deprecate assignAt as well in favor of the new copyFrom?  
I think the new name is clearer (and matches a similar method in Image).  The only place we had used assignAt was in the old Sum._shoot method, so that's now gone with the new algorithm.  And I doubt any users probably use assignAt for anything.

**Comment by jmeyers314** [comment]
> Do you think we should deprecate assignAt as well in favor of the new copyFrom?

Yep, this sounds good to me.
